superset-frontend/src/views/routes.tsx

```diff
@@ -189,6 +197,14 @@ export const routes: Routes = [
     path: '/superset/explore/p',
     Component: ExplorePage,
   },
+  {
+    path: '/dataset/add/',
+    Component: DatasetEditor,
+  },
+  {
+    path: '/dataset/:datasetId',
+    Component: DatasetEditor,
```

Comment:
do you think these two pages are similar enough to share the same parent component?

Reply:
No this was based on a previous understanding of the code, I changed it to reflect the components more accurately. 

I did reuse the header component, so I moved it one level higher.